source/adapters/level_zero/enqueue_native.cpp

```diff
@@ -10,11 +10,57 @@
 
 #include <ur_api.h>
 
+#include "logger/ur_logger.hpp"
 #include "queue.hpp"
+#include "ur_level_zero.hpp"
 
 ur_result_t ur_queue_handle_legacy_t_::enqueueNativeCommandExp(
-    ur_exp_enqueue_native_command_function_t, void *, uint32_t,
-    const ur_mem_handle_t *, const ur_exp_enqueue_native_command_properties_t *,
-    uint32_t, const ur_event_handle_t *, ur_event_handle_t *) {
-  return UR_RESULT_ERROR_UNSUPPORTED_FEATURE;
+    ur_exp_enqueue_native_command_function_t pfnNativeEnqueue, void *data,
```

Comment:
for now this implementation can be literally a copy/paste from:
https://github.com/oneapi-src/unified-runtime/blob/main/source/adapters/level_zero/memory.cpp#L1203

just replace:
`  ZE2UR_CALL(zeCommandListAppendMemoryPrefetch, (ZeCommandList, Mem, Size));`
with:
`pfnNativeEnqueue(Queue, data);`

`Queue->queueFinish();` should not be necessary. Please sync with @kswiecicki if you need help.

Reply:
Thanks @pbalcer that is great